<summary>
    <font size="8" color="palevioletred"> El índice de marginalización de México </font>
</summary>
<summary>
    <font size="3" > Proyecto final de programación </font>
</summary>

---
<div style="text-align: center;">
    <img src="https://github.com/estephaniapa/PropedeuticoRedes/blob/main/A%CC%81lgebra%20lineal%20y%20optimizacio%CC%81n/mcd.png?raw=true" alt="Image" style="float: right; width: 150px;">
</div>

<summary>
    <font size="5" color="yellowgreen"> Maestría en Ciencias de Datos </font>
</summary>
<summary>
    <font size="5" color="yellowgreen"> Universidad de Sonora </font>
</summary>
<br>
<summary>
    <font size="4"> Propedeutico: Programación </font> 
</summary>
<summary>
    <font size="4"> Profesor: Dr. Julio Waissman Vilanova </font>
</summary>
<summary>
    <font size="4"> Alumna: Estephania Pivac Alcaraz </font>
</summary>

---

## Introducción

## 1. Descarga y lectura de los datos

Descargamos y leemos el archivo de la base de datos de índice de marginalización por Municipio 2020. Los datos se obtuvieron de la página de conapo del gobierno de México.

In [1]:
# Importamos las librerías necesarias
import pandas as pd
import plotly.graph_objects as go
import plotly.express as px
import plotly.io as pio
import matplotlib.pyplot as plt
import numpy as np

In [2]:
# Cargamos los datos
marginacion_df= pd.read_excel('http://www.conapo.gob.mx/work/models/CONAPO/Marginacion/Datos_Abiertos/Municipio/IMM_2020.xls', sheet_name='IMM_2020')

In [3]:
# Exploramos los datos
marginacion_df.head()

CVE_ENT         NOM_ENT  CVE_MUN         NOM_MUN  POB_TOT     ANALF   
0        1  Aguascalientes     1001  Aguascalientes   948990  1.644738  \
1        1  Aguascalientes     1002        Asientos    51536  3.526405   
2        1  Aguascalientes     1003        Calvillo    58250  4.491509   
3        1  Aguascalientes     1004           Cosío    17000  3.144867   
4        1  Aguascalientes     1005     Jesús María   129929  2.380588   

       SBASC     OVSDE     OVSEE     OVSAE      OVPT       VHAC    PL.5000   
0  20.367220  0.104799  0.113169  0.378610  0.591434  10.339530   7.523683  \
1  33.906364  2.650373  0.486448  0.858160  1.352430  22.942305  78.221049   
2  42.482450  0.365177  0.516760  0.800978  1.040411  19.219858  51.301288   
3  27.696745  0.712855  0.577354  0.659833  1.030989  22.716866  65.470588   
4  26.692477  0.277034  0.354957  0.860426  1.312652  16.404575  37.164143   

       PO2SM    IM_2020   GM_2020  IMN_2020  
0  54.226594  60.318795  Muy bajo  0.944508  
1  78.565471  56.546071  Muy bajo  0.885433  
2  79.259777  57.058251  Muy bajo  0.893453  
3  81.726369  57.114030  Muy bajo  0.894326  
4  56.748753  59.011762  Muy bajo  0.924042

## 2. Descripción del DataFrame

La base de datos está compuesta por 17 columnas y 2469 renglones. A continuación mostramos el diccionario de datos.

<t align="center">

| Variable | Descripción                                                             |
|----------|-------------------------------------------------------------------------|
| CVE_ENT  | Clave de entidad federativa                                             |
| NOM_ENT  | Nombre de entidad federativa                                            |
| CVE_MUN  | Clave del municipio                                                     |
| NOM_MUN  | Nombre del municipio                                                    |
| POB_TOT  | Población total                                                         |
| ANALF    | Porcentaje de población analfabeta de 15 años o más                     |
| SBASC    | Porcentaje de población de 15 años o más sin educación básica            |
| OVSDE    | Porcentaje de ocupantes en viviendas particulares sin drenaje ni excusado|
| OVSEE    | Porcentaje de ocupantes en viviendas particulares sin energía eléctrica  |
| OVSAE    | Porcentaje de ocupantes en viviendas particulares sin agua entubada      |
| OVPT     | Porcentaje de ocupantes en viviendas particulares con piso de tierra     |
| VHAC     | Porcentaje de viviendas particulares con hacinamiento                   |
| PL.5000  | Porcentaje de población que vive en localidades menores a 5 000 habitantes|
| PO2SM    | Porcentaje de población ocupada con ingresos de hasta 2 salarios mínimos|
| IM_2020  | Índice de marginación, 2020                                             |
| GM_2020  | Grado de marginación, 2020                                              |
| IMN_2020 | Índice de marginación normalizado, 2020                                  |



In [4]:
marginacion_df.info()

<class 'pandas.core.frame.DataFrame'>
RangeIndex: 2469 entries, 0 to 2468
Data columns (total 17 columns):
 #   Column    Non-Null Count  Dtype  
---  ------    --------------  -----  
 0   CVE_ENT   2469 non-null   int64  
 1   NOM_ENT   2469 non-null   object 
 2   CVE_MUN   2469 non-null   int64  
 3   NOM_MUN   2469 non-null   object 
 4   POB_TOT   2469 non-null   int64  
 5   ANALF     2469 non-null   float64
 6   SBASC     2469 non-null   float64
 7   OVSDE     2469 non-null   float64
 8   OVSEE     2469 non-null   float64
 9   OVSAE     2469 non-null   float64
 10  OVPT      2469 non-null   float64
 11  VHAC      2469 non-null   float64
 12  PL.5000   2469 non-null   float64
 13  PO2SM     2469 non-null   float64
 14  IM_2020   2469 non-null   float64
 15  GM_2020   2469 non-null   object 
 16  IMN_2020  2469 non-null   float64
dtypes: float64(11), int64(3), object(3)
memory usage: 328.0+ KB


In [5]:
marginacion_df.describe()

CVE_ENT       CVE_MUN       POB_TOT        ANALF        SBASC   
count  2469.000000   2469.000000  2.469000e+03  2469.000000  2469.000000  \
mean     19.215472  19324.164844  5.103849e+04    10.164466    45.853026   
std       7.367340   7382.892458  1.469907e+05     7.633633    13.981594   
min       1.000000   1001.000000  8.100000e+01     0.353446     5.535137   
25%      14.000000  14079.000000  4.489000e+03     4.427755    35.737568   
50%      20.000000  20226.000000  1.355200e+04     8.202762    46.339439   
75%      24.000000  24027.000000  3.528400e+04    13.787294    55.856378   
max      32.000000  32058.000000  1.922523e+06    53.071253    88.328076   

             OVSDE        OVSEE        OVSAE         OVPT         VHAC   
count  2469.000000  2469.000000  2469.000000  2469.000000  2469.000000  \
mean      3.159963     1.500793     6.118145     7.987232    26.566286   
std       5.289299     2.769167     9.245995     8.973591    10.586540   
min       0.000000     0.000000     0.000000     0.000000     3.950392   
25%       0.651869     0.366077     0.878499     1.654653    18.725100   
50%       1.428250     0.828157     2.452316     4.714141    25.000000   
75%       3.342618     1.678328     7.285869    11.029646    32.820816   
max      64.450424    53.065463    81.788441    68.149764    69.564018   

           PL.5000        PO2SM      IM_2020     IMN_2020  
count  2469.000000  2469.000000  2469.000000  2469.000000  
mean     69.900469    82.143854    53.955581     0.844869  
std      35.267726    11.830444     3.904590     0.061140  
min       0.000000    28.453113    21.406635     0.335198  
25%      40.129696    74.615600    51.844432     0.811812  
50%     100.000000    84.643266    54.423506     0.852196  
75%     100.000000    91.620112    56.696126     0.887782  
max     100.000000   100.000000    62.397145     0.977052

In [6]:
marginacion_df.describe(include=object)

NOM_ENT        NOM_MUN   GM_2020
count     2469           2469      2469
unique      32           2328         5
top     Oaxaca  Benito Juárez  Muy bajo
freq       570              7       655

Al explorar estos datos hubieron varios puntos que llamaron mi atención:

1. El promedio del porcentaje de población de 15 años o más sin educación básica (SBASC) es del 45.8%, lo cual me resulta increiblemente alto.

2. El porcentaje mas alto de analfabetismo (ANALF) registrado fue de 53%, por lo que existe al menos un municipio donde la mayor parte de su población es analfabeta.

3. El porcentaje promedio de personas que viven en localidades menores a 5000 (PL.5000) habitantes es de 69.9%, por lo que en promedio la mayoria de la población por municipio vive en localidades muy pequeñitas.

4. El porcentaje promedio de personas ocupadas con ingresos de hasta 2 salarios mínimos es del 82.14%, por lo que la mayoría de la población ocupada tienen salarios muy bajos.

5. El grado de marginación más repetido fue 'muy bajo'. De aquí podemos concluir que la mayoría de los municipios gozan de un bajo nivel de marginación. 

## 3. Análisis de Marginalización: Porcentaje de Municipios por Estado

La siguiente gráfica muestra el porcentaje de municipios por estado con grado de marginación 
- "muy bajo"
- "bajo"
- "medio"
- "alto"
- "muy alto"

Esto permite realizar un análisis de la marginación y desigualdad por estados.

In [7]:
# Creamos un dataframe con la información relevante para la gráfica
distribucion_df = marginacion_df.groupby(['NOM_ENT','GM_2020'])[['NOM_MUN']].count().reset_index()

# Calculamos el total de municipios por estados para poder calcular el porcentaje relativo
group_totals = distribucion_df.groupby('NOM_ENT')['NOM_MUN'].transform('sum')

# Calcula el porcentaje relativo, esto es, cantidad de municipios en cada categiría de índice sobre el total de municipios
distribucion_df['Porcentaje'] = (distribucion_df['NOM_MUN'] / group_totals) * 100

distribucion_df.head()

NOM_ENT   GM_2020  NOM_MUN  Porcentaje
0       Aguascalientes  Muy bajo       11  100.000000
1      Baja California     Medio        1   16.666667
2      Baja California  Muy bajo        5   83.333333
3  Baja California Sur  Muy bajo        5  100.000000
4             Campeche      Alto        2   16.666667

In [8]:
#Crear la gráfica
fig = px.bar(distribucion_df, x='NOM_ENT', y='Porcentaje', color='GM_2020', hover_name='GM_2020',
             title='Porcentaje de municipios por estado por grado de marginación',
             category_orders={'GM_2020': ['Muy alto', 'Alto', 'Medio', 'Bajo', 'Muy bajo']})
# Editar título del eje x
fig.update_xaxes(title_text='Estado')
# Editar título del eje y
fig.update_yaxes(title_text='Porcentaje de municipios')
# Editar título de la leyenda
fig.update_layout(legend_title_text='Grado de marginación')
# Editar tooltips
fig.update_traces(hovertemplate='<b>%{x}</b><br>' +
                                'Grado de marginación: %{hovertext}<br>' +
                                'Porcentaje de municipios: %{y:.1f}%')
# Personalizar el acomodo de las etiquetas del eje X y la altura de la gráfica
fig.update_layout(
    xaxis=dict(tickangle=315),
    height = 600
)
# Guardamos la gráfica como una imagen estática en formato png
pio.write_image(fig, 'grafica1.png')

# Imprimimos la gráfica
fig.show()

In [9]:
# # Agrupa tus datos por 'NUM_ENT' y 'GMM_2020', y calcula la suma de 'Porcentaje' para cada grupo
# grouped = distribucion_df.groupby(['NOM_ENT', 'GM_2020'])['Porcentaje'].mean().unstack()

# # Crea una gráfica de barras apiladas
# grouped.plot(kind='bar', stacked=True)

# # Muestra la gráfica
# plt.show()

## 4. Análisis de Marginación: Porcentaje de Población por Estado

In [10]:
# Creamos el dataframe con la información agrupada que se requiere para las gráficas
distribucion_poblacion_df = marginacion_df.groupby(['NOM_ENT','GM_2020'])[['POB_TOT']].sum().reset_index()

# Calculamos el total de población por estado
total_por_estado = distribucion_poblacion_df.groupby('NOM_ENT')['POB_TOT'].transform('sum')

# Calculamos el porcentaje relativo de la distribución de la población
distribucion_poblacion_df['Porcentaje'] = (distribucion_poblacion_df['POB_TOT'] / total_por_estado) * 100

# Renombramos las columnas
distribucion_poblacion_df = distribucion_poblacion_df.rename(columns={'NOM_ENT':'Estado', 'GM_2020': 'Grado de marginación', 'POB_TOT':'Población'})
distribucion_poblacion_df.head()

Estado Grado de marginación  Población  Porcentaje
0       Aguascalientes             Muy bajo    1425607  100.000000
1      Baja California                Medio     117568    3.119325
2      Baja California             Muy bajo    3651452   96.880675
3  Baja California Sur             Muy bajo     798447  100.000000
4             Campeche                 Alto      78627    8.469424

In [11]:
# Creamos la gráfica dinámica en plotly
fig = px.bar(distribucion_poblacion_df, x='Estado', y='Porcentaje', color='Grado de marginación', hover_name='Grado de marginación',
             title='Porcentaje de población por estado por grado de marginación',
             category_orders={'Grado de marginación': ['Muy alto', 'Alto', 'Medio', 'Bajo', 'Muy bajo']})
# Editamos título del eje y
fig.update_yaxes(title_text='Porcentaje de municipios')
# Editamos tooltips
fig.update_traces(hovertemplate='<b>%{x}</b><br>' +
                                'Grado de marginación: %{hovertext}<br>' +
                                'Porcentaje de población: %{y:.1f}%')
# Personalizamos la ubicación de las etiquetas del eje X y el tamaño de la gráfica
fig.update_layout(
    xaxis=dict(tickangle=315),
    height = 600
)
# Guardamos la gráfica en formato jpg
pio.write_image(fig, 'grafica2.jpg')
fig.show()

In [12]:
# # Agrupa tus datos por 'NUM_ENT' y 'GMM_2020', y calcula la suma de 'Porcentaje' para cada grupo
# grouped = marginacion_df.groupby(['NOM_ENT', 'GM_2020'])['POB_TOT'].sum().unstack()

# # Crea una gráfica de barras apiladas
# grouped.plot(kind='bar', stacked=True)

# # Muestra la gráfica
# plt.show()

## 5. Análisis de marginación y desigualdad
<t align="justify">

**¿Hay coincidencias entra la gráficas anteriores?  ¿Algún hallazgo? Comenta tu análisis**

Las gráficas anteriores presentan un análisis del grado de marginación estatal en el país desde dos perspectivas distintas, proporcionando así una visión más completa de la situación. Aunque los resultados difieren en cierta medida, existe una relación entre ambas perspectivas que nos permite comprender mejor la complejidad de la marginación en México.

Ambas gráficas tienen ciertas similitudes, por ejemplo: 
- Los estados que identificamos con niveles más altos de algún grado de marginación se mantienen en ambas gráficas, pero pareciera que la primera gráfica podría resultar algo engañosa, ya que cuando analizamos la marginación en porcentajes poblacionales notamos que en la mayoría de los estados el nivel de marginación no es tan alto. 
- Un estado que me impresionó mucho fue Chihuahua, donde parecía que tenía niveles de marginación muy altos (seguramente por su gran extensión territorial, y gran cantidad de regiones rurales) pero al analizarlo poblacionalmente el porcentaje es mucho más bajo.


### Gráfica 1

A partir del análisis de la gráfica, se destaca de manera significativa el estado de Guerrero, el cual exhibe el porcentaje más alto de municipios con un grado de marginación muy alto, alcanzando un preocupante 42%. Esta cifra revela una situación alarmante, ya que casi la mitad de la población en dicho estado enfrenta un nivel significativo de desigualdad socioeconómica. 

Asimismo, se observa que los estados de Chihuahua, Chiapas, Oaxaca y Nayarit también presentan porcentajes elevados de municipios con grado de marginación muy alto, con valores muy similares entre ellos, registrando un 16.4%, 16.1%, 16.1% y 15% respectivamente. Estos datos demuestran la persistencia de condiciones desfavorables y desafíos en términos de desarrollo humano en dichas regiones.

En cuanto al grado de marginación "Alto", se evidencia que este nivel se encuentra más extendido en todos los estados analizados. Guerrero, Chiapas y Oaxaca continúan mostrando los valores más altos en esta categoría, lo cual subraya la situación desfavorable que enfrentan estas regiones en términos de desigualdad y exclusión social. Es importante destacar que estos tres estados en conjunto concentran más del 50% de los municipios con un grado alto y muy alto de marginación, lo que indica la urgencia de implementar políticas y acciones para abordar estas disparidades y fomentar un desarrollo más equitativo y sostenible en estas áreas.

En resumen, los resultados obtenidos a partir de la gráfica revelan un panorama preocupante en cuanto a la desigualdad socioeconómica y el grado de marginación en los municipios de diversos estados. Estos hallazgos subrayan la necesidad de adoptar medidas efectivas y dirigidas para reducir las brechas de desigualdad, promover el desarrollo inclusivo y mejorar las condiciones de vida de las poblaciones en situación de marginación en todo el país.

Además, es importante destacar que el estado de Sonora se distingue en la gráfica por presentar un porcentaje notablemente bajo de municipios con un grado de marginación "Alto". 


### Gráfica 2

En la segunda gráfica, realizamos una comparación del porcentaje de población estatal que se encuentra en distintos grados de marginación. Nos enfocaremos en aquellos estados que presentan un mayor grado de marginación alta y muy alta, para obtener una comprensión más detallada de la situación.

Es importante resaltar que esta gráfica muestra porcentajes más bajos en comparación con la anterior. Esto tiene sentido, ya que es probable que los municipios con mayor nivel de marginación se encuentren en áreas rurales y con menor concentración de población. Personalmente, esperaba que estos porcentajes fueran aún más bajos.

En el caso específico de Guerrero, continúa presentando el porcentaje más alto (23.4%) de población en el nivel de marginación muy alta, seguido de cerca por Chiapas y Oaxaca con porcentajes de 18.8% y 14.4%, respectivamente. Aunque los valores no son tan altos como en la gráfica anterior, existe una diferencia considerable con respecto a los niveles de marginación de Chihuahua y Nayarit, que muestran un menor porcentaje de población en esta categoría de marginación extrema.

Al analizar el grado de marginación alto, observamos que Chiapas lidera con el porcentaje más alto de población, alcanzando el 31%. Esto implica que casi la mitad de la población en este estado se encuentra en niveles de marginación alta y muy alta. Es preocupante ver cómo una proporción significativa de la población chiapaneca enfrenta condiciones de desigualdad extrema.

Además, Guerrero ocupa el segundo lugar en desigualdad, seguido de cerca por Oaxaca, con porcentajes de 47% y 42% respectivamente. Estos estados muestran un panorama alarmante en términos de marginación y desigualdad socioeconómica. Los altos porcentajes de población en condiciones precarias resaltan la urgencia de implementar políticas y acciones que aborden la marginación de manera integral y promuevan una distribución más equitativa de los recursos y oportunidades en estas regiones.

### Conclusión

Considero que la gráfica más adecuada para analizar los niveles de marginación es la segunda, donde medimos porcentajes poblacionales.

## 6. Porcentaje de analfabetismo vs. porcentaje de poblaciones en localidades de menos de 5,000 habitantes
Grafica la relación de porcentaje de analfabetismo respecto al porcentaje de poblaciones en localidades de menos de 5,000 habitantes.

In [13]:
fig = px.scatter(marginacion_df, x="PL.5000", y="ANALF")
fig.show()

<t align="justify">

A partir de la gráfica podemos concluir que no parece existir una relación evidente entre estas dos variables, ya que no se identifica un patron de comportamiento. Quisiera destacar que la dispersion de la distribución de analfabetismo aumenta a medida que el porcentaje de población en localidades de menor a 5000 personas aumenta. Uno esperaría que las localidades muy pequeñas presentaran valores mayores de analfabetismo, pero los datos no parecen evidenciar esta hipótesis.

## 7. Análisis de correlación 
<t align="justify">

Para poder identificar las variables que tienen una mayor correlación con el porcentaje de analfabetismo en personas mayores a 15 años debemos calcular una matriz de correlación que incluya todas las variables en cuestión.

Recordemos que la matriz de correlación es una herramienta estadística que permite analizar la relación entre variables. Se utiliza para determinar la fuerza y la dirección de la asociación entre pares de variables en un conjunto de datos. Esta matriz se representa como una tabla cuadrada donde cada celda contiene un valor que indica el grado de correlación entre dos variables.

La correlación es una medida estadística que cuantifica la relación lineal entre dos variables. Puede tomar valores en el rango de -1 a 1. Un valor de 1 indica una correlación positiva perfecta, lo que significa que a medida que una variable aumenta, la otra también lo hace de manera proporcional. Por otro lado, un valor de -1 indica una correlación negativa perfecta, lo que significa que a medida que una variable aumenta, la otra disminuye en forma proporcional. Un valor de 0 indica una ausencia de correlación, es decir, no hay relación lineal entre las variables.

Para calcular cada entrada de la matriz de correlación, se utiliza un coeficiente de correlación, el más comúnmente utilizado es el **coeficiente de correlación de Pearson**. Este coeficiente se calcula dividiendo la covarianza de las dos variables por el producto de las desviaciones estándar de cada variable. La covarianza mide cómo varían conjuntamente las dos variables, mientras que las desviaciones estándar miden la dispersión de cada variable por separado. El coeficiente de correlación de Pearson varía entre -1 y 1, donde un valor cercano a 1 o -1 indica una correlación fuerte, y un valor cercano a 0 indica una correlación débil o nula.

La matriz de correlación es una herramienta fundamental en el análisis de datos, ya que permite identificar patrones, dependencias y tendencias entre variables. Al examinar la matriz de correlación, es posible determinar qué variables están fuertemente relacionadas entre sí y cuáles pueden considerarse independientes. Esto proporciona información valiosa para el análisis que deseamos realizar.

In [14]:
# Seleccionamos las variables de interés
marginacion_num_df = marginacion_df[['POB_TOT', 'ANALF', 'SBASC',
       'OVSDE', 'OVSEE', 'OVSAE', 'OVPT', 'VHAC', 'PL.5000', 'PO2SM',
       'IM_2020']]

In [15]:
# Calculamos la matriz de correlación

matriz_corr = marginacion_num_df.corr()
matriz_corr


POB_TOT     ANALF     SBASC     OVSDE     OVSEE     OVSAE      OVPT   
POB_TOT  1.000000 -0.236801 -0.387466 -0.111017 -0.105180 -0.092102 -0.161628  \
ANALF   -0.236801  1.000000  0.779053  0.347537  0.351523  0.360644  0.644960   
SBASC   -0.387466  0.779053  1.000000  0.259337  0.296920  0.269822  0.548058   
OVSDE   -0.111017  0.347537  0.259337  1.000000  0.529746  0.219278  0.279222   
OVSEE   -0.105180  0.351523  0.296920  0.529746  1.000000  0.343723  0.501319   
OVSAE   -0.092102  0.360644  0.269822  0.219278  0.343723  1.000000  0.391056   
OVPT    -0.161628  0.644960  0.548058  0.279222  0.501319  0.391056  1.000000   
VHAC    -0.206148  0.658532  0.521014  0.374002  0.308139  0.319650  0.615184   
PL.5000 -0.422541  0.449324  0.598548  0.205200  0.231066  0.189617  0.338568   
PO2SM   -0.410937  0.620011  0.702248  0.204174  0.197733  0.276074  0.477667   
IM_2020  0.315036 -0.847822 -0.768872 -0.566430 -0.636501 -0.581819 -0.772474   

             VHAC   PL.5000     PO2SM   IM_2020  
POB_TOT -0.206148 -0.422541 -0.410937  0.315036  
ANALF    0.658532  0.449324  0.620011 -0.847822  
SBASC    0.521014  0.598548  0.702248 -0.768872  
OVSDE    0.374002  0.205200  0.204174 -0.566430  
OVSEE    0.308139  0.231066  0.197733 -0.636501  
OVSAE    0.319650  0.189617  0.276074 -0.581819  
OVPT     0.615184  0.338568  0.477667 -0.772474  
VHAC     1.000000  0.269036  0.580601 -0.731332  
PL.5000  0.269036  1.000000  0.575018 -0.586803  
PO2SM    0.580601  0.575018  1.000000 -0.699429  
IM_2020 -0.731332 -0.586803 -0.699429  1.000000

A partir de esta matriz es muy dificil sacar conclusiones, incluso distinguir el valor mayor es complicado. Por lo que, para realizar el análisis de correlación graficaremos la matriz de correlación en un heatmap.

In [16]:
# Creamos un heatmap para poder visualizar los valores de los coeficientes de correlación
fig = px.imshow(matriz_corr.values,
                x=matriz_corr.columns,
                y=matriz_corr.index,
                title='Matriz de correlación')

# Personalizamos el tamaño del heatmap
fig.update_layout(width=600,height=600)

# Graficamos
fig.show()

<t align="justify">

A partir de este heatmap, podemos observar las principales variables correlacionadas con el porcentaje de analfabetismo en la población:

1. **Índice de marginación:** Como era de esperar, el índice de marginación muestra una correlación significativa con el porcentaje de analfabetismo. Esto se debe a que el índice de marginación incluye el porcentaje de analfabetismo como uno de sus indicadores, por lo que existe una relación directa entre ambas variables.

2. **Porcentaje de población de 15 años o más sin educación básica (SBASC):** Existe una correlación alta y positiva de 0.77 entre esta variable y el porcentaje de analfabetismo. Esto indica que a medida que aumenta la población sin educación básica, también tiende a aumentar el porcentaje de analfabetismo.

3. **Porcentaje de viviendas particulares con hacinamiento (VHAC):** Se observa una correlación mediana y positiva de 0.65 entre el hacinamiento y el analfabetismo. El hacinamiento se caracteriza por la falta de suficiente espacio habitacional adecuado, lo cual dificulta las condiciones propicias para el estudio y puede contribuir a un mayor nivel de analfabetismo.

4. **Porcentaje de ocupantes en viviendas particulares habitadas con piso de tierra (OVPT):** Con un coeficiente de correlación de 0.64, se evidencia una relación positiva y mediana entre esta variable y el analfabetismo. Esta característica también está relacionada con condiciones de pobreza, y su presencia puede estar asociada a un mayor nivel de analfabetismo.

5. **Porcentaje de población ocupada con ingresos de hasta 2 salarios mínimos (PO2SM):** Se identifica una correlación positiva de magnitud mediana (coeficiente de 0.62) entre esta variable y el analfabetismo. Aunque la correlación no es fuerte, indica que existe una relación directa entre la proporción de población con ingresos bajos y el porcentaje de analfabetismo. Esta variable también refleja condiciones económicas desfavorables que pueden dificultar el acceso a la educación.

Estas correlaciones resaltan la importancia de abordar la educación y la pobreza en conjunto, ya que están estrechamente relacionadas. Implementar políticas y programas que fomenten la educación inclusiva y mejoren las condiciones socioeconómicas puede ser fundamental para reducir el nivel de analfabetismo y promover un desarrollo equitativo en la sociedad.

## 8. Propuesta de indicadores: Índice de Marginación de Vivienda

<t align="justify">

El Índice de Marginación en Vivienda es un indicador que se crea a partir de la combinación de diversas variables relacionadas con las condiciones de vivienda en una determinada área geográfica. Este índice permite evaluar la calidad de las condiciones de vivienda y reflejar el grado de marginación en este aspecto.

El índice se construye utilizando los siguientes porcentajes de viviendas:

- **Porcentaje de viviendas sin drenaje ni excusado (OVSDE):** Este indicador mide la falta de acceso a servicios básicos de saneamiento en las viviendas.

- **Porcentaje de viviendas sin energía eléctrica (OVSEE):** Este indicador muestra la falta de acceso a la electricidad en las viviendas.

- **Porcentaje de viviendas sin agua entubada (OVSAE):** Este indicador refleja la falta de acceso a agua potable a través de una red de distribución en las viviendas.

- **Porcentaje de viviendas con piso de tierra (OVPT):** Este indicador indica la ausencia de un piso adecuado en las viviendas, lo cual puede ser un indicio de condiciones precarias de vivienda.

- **Porcentaje de viviendas con hacinamiento (VHAC):** Este indicador muestra la proporción de viviendas donde existe una alta densidad de ocupantes en relación con el espacio disponible, lo cual puede ser un indicador de condiciones de hacinamiento.

Para calcular el Índice de Marginación en Vivienda, se pueden asignar pesos a cada una de estas variables según su importancia relativa en la evaluación de las condiciones de vivienda. Estos pesos pueden ser determinados mediante métodos estadísticos o mediante la opinión de expertos en el tema. Una vez asignados los pesos, se realiza una combinación ponderada de los porcentajes de viviendas en cada variable para obtener el índice.

Este índice proporciona una medida agregada de la marginación en el ámbito de la vivienda en una determinada área geográfica, y puede ser utilizado como una herramienta para identificar áreas con condiciones de vivienda precarias y orientar la implementación de políticas y programas de mejoramiento de vivienda en esas áreas.

Los pesos para este indicador los asignaré de manera proporcional a la correlación que existe con el nivel de analfabetismo. Este indicador podría resultar de utilidad para la creación de estrategias y políticas públicas orientadas a disminuir los niveles de analfabetismo en los sectores que presentan niveles muy altos de marginación de vivienda.

In [17]:
# Seleccionamos los coeficientes de correlación
pesos = matriz_corr.loc['ANALF', ['OVSDE', 'OVSEE', 'OVSAE', 'OVPT', 'VHAC']]

# Normalizamos los pesos para que sumen 1 
pesos = np.divide(pesos, pesos.sum())
pesos

OVSDE    0.147062
OVSEE    0.148749
OVSAE    0.152609
OVPT     0.272919
VHAC     0.278662
Name: ANALF, dtype: float64

In [18]:
# Creamos el nuevo dataframe con este indicador
marginacion_df_nuevo = marginacion_df.loc[:,['CVE_ENT', 'NOM_ENT', 'CVE_MUN', 'NOM_MUN', 'POB_TOT', 'ANALF', 'PL.5000', 'PO2SM',
       'IM_2020', 'GM_2020', 'IMN_2020']]
marginacion_df_nuevo.loc[:,'IMV_2020'] = (pesos[0]*marginacion_df['OVSDE'] + 
                                          pesos[1]*marginacion_df['OVSEE'] + 
                                          pesos[2]*marginacion_df['OVSAE'] + 
                                          pesos[3]*marginacion_df['OVPT'] +
                                          pesos[4]*marginacion_df['VHAC']                             
)
marginacion_df_nuevo.head()

CVE_ENT         NOM_ENT  CVE_MUN         NOM_MUN  POB_TOT     ANALF   
0        1  Aguascalientes     1001  Aguascalientes   948990  1.644738  \
1        1  Aguascalientes     1002        Asientos    51536  3.526405   
2        1  Aguascalientes     1003        Calvillo    58250  4.491509   
3        1  Aguascalientes     1004           Cosío    17000  3.144867   
4        1  Aguascalientes     1005     Jesús María   129929  2.380588   

     PL.5000      PO2SM    IM_2020   GM_2020  IMN_2020  IMV_2020  
0   7.523683  54.226594  60.318795  Muy bajo  0.944508  3.132668  
1  78.221049  78.565471  56.546071  Muy bajo  0.885433  7.355333  
2  51.301288  79.259777  57.058251  Muy bajo  0.893453  5.892590  
3  65.470588  81.726369  57.114030  Muy bajo  0.894326  6.903104  
4  37.164143  56.748753  59.011762  Muy bajo  0.924042  5.154421

In [19]:
marginacion_df_nuevo.loc[:,'IMV_2020'].describe()

count    2469.000000
mean       11.204500
std         6.122139
min         1.125172
25%         6.621967
50%        10.094561
75%        14.201953
max        49.145384
Name: IMV_2020, dtype: float64

In [20]:
# Veremos la distribución del índice de marginación de vivienda
df = px.data.tips()
fig = px.histogram(marginacion_df_nuevo, x="IMV_2020", nbins=50, title='Distribución del Índice de Marginación de Vivienda municipal',)
# Editar título del eje x
fig.update_xaxes(title_text='Valores del Índice de Marginación de Vivienda (IMV_2020)')
# Editar título del eje y
fig.update_yaxes(title_text='Cantidad de municipios')

fig.show()

## 9. Conclusión
<t align="justify">

En este proyecto, se llevó a cabo un análisis detallado del índice de marginación en México, utilizando datos sociodemográficos a nivel municipal. Se exploraron diversas variables relacionadas con la marginación, como el analfabetismo, la falta de educación básica, las condiciones de vivienda precarias y los ingresos bajos.

A través del análisis de correlación, se identificaron las variables más significativas que están asociadas con altos niveles de marginación. Estas variables incluyen el índice de marginación, el porcentaje de población sin educación básica, las condiciones deficientes de vivienda (como la falta de servicios básicos) y los bajos ingresos.

La visualización de los datos en forma de gráficos y mapas permitió una comprensión más clara de la distribución geográfica de la marginación en el país. Se observó que ciertos estados, como Guerrero, Chiapas y Oaxaca, presentan niveles alarmantes de marginación, con un alto porcentaje de su población viviendo en condiciones desfavorables.

Estos resultados resaltan la necesidad de implementar políticas y acciones enfocadas en reducir la marginación y promover la equidad en el acceso a la educación, vivienda adecuada y oportunidades económicas. El desarrollo de indicadores como el Índice de Marginación en Vivienda propuesto proporciona una herramienta valiosa para evaluar la calidad de vida de las comunidades y orientar los esfuerzos de intervención.

## 10. Bibliografía

- Consejo Nacional de Población (CONAPO). (2020). Índice de marginación por entidad federativa y municipio 2020. Recuperado de [https://www.gob.mx/cms/uploads/attachment/file/685354/Nota_te_cnica_IMEyM_2020.pdf]

- Consejo Nacional de Población (CONAPO). (2020). Índices de Marginación 2020. Recuperado de [https://www.gob.mx/conapo/documentos/indices-de-marginacion-2020-284372]